crates/router/src/connector/forte/transformers.rs

```diff
@@ -63,62 +66,93 @@ impl TryFrom<utils::CardIssuer> for ForteCardType {
-                let address = item.get_billing_address()?;
+                let card_type = ForteCardType::try_from(card.get_card_issuer()?)?;
+                let address = item.router_data.get_billing_address()?;
                 let card = Card {
                     card_type,
-                    name_on_card: ccard.card_holder_name.clone(),
-                    account_number: ccard.card_number.clone(),
-                    expire_month: ccard.card_exp_month.clone(),
-                    expire_year: ccard.card_exp_year.clone(),
-                    card_verification_value: ccard.card_cvc.clone(),
+                    name_on_card: card.card_holder_name.clone(),
+                    account_number: card.card_number.clone(),
+                    expire_month: card.card_exp_month.clone(),
+                    expire_year: card.card_exp_year.clone(),
+                    card_verification_value: card.card_cvc.clone(),
                 };
                 let billing_address = BillingAddress {
                     first_name: address.get_first_name()?.to_owned(),
                     last_name: address.get_last_name()?.to_owned(),
                 };
-                let authorization_amount =
-                    utils::to_currency_base_unit_asf64(item.request.amount, item.request.currency)?;
+                let authorization_amount = utils::to_currency_base_unit_asf64(
+                    item.router_data.request.amount,
+                    item.router_data.request.currency,
+                )?;
                 Ok(Self {
                     action,
                     authorization_amount,
                     billing_address,
                     card,
                 })
             }
-            api_models::payments::PaymentMethodData::CardRedirect(_)
-            | api_models::payments::PaymentMethodData::Wallet(_)
-            | api_models::payments::PaymentMethodData::PayLater(_)
-            | api_models::payments::PaymentMethodData::BankRedirect(_)
-            | api_models::payments::PaymentMethodData::BankDebit(_)
-            | api_models::payments::PaymentMethodData::BankTransfer(_)
-            | api_models::payments::PaymentMethodData::Crypto(_)
-            | api_models::payments::PaymentMethodData::MandatePayment {}
-            | api_models::payments::PaymentMethodData::Reward {}
-            | api_models::payments::PaymentMethodData::Upi(_)
-            | api_models::payments::PaymentMethodData::Voucher(_)
-            | api_models::payments::PaymentMethodData::GiftCard(_) => {
-                Err(errors::ConnectorError::NotSupported {
-                    message: utils::SELECTED_PAYMENT_METHOD.to_string(),
-                    connector: "Forte",
-                })?
+            _ => {
+                return Err(
+                    errors::ConnectorError::NotImplemented("Payment methods".to_string()).into(),
+                )
```

Comment:
Please don't add this change, these types are intentionally added in the match statement so that all cases are covered exhaustively and errors can be known at compile time